algorithms/rainbow/model.py

```diff
@@ -0,0 +1,112 @@
+"""
+Model definition definition for DQN and Noisy layers
+"""
+import math
+import torch
+from torch import nn
+from torch.nn import functional as F
+
+
+class RainbowDQN(nn.Module):
+    """
+    From the paper "Rainbow: Combining Improvements in Deep Reinforcement Learning"
+    Source: https://arxiv.org/pdf/1710.02298.pdf
+
+    Main model used as Q function estimator for policy generation
+    """
+    def __init__(self, args, action_space):
+        super().__init__()
+        self.atoms = args.atoms
+        self.action_space = action_space.n
+        self.linear_in = self.get_linear_size(args)
+
+        self.conv1 = nn.Conv2d(args.history_length, 32, 8, stride=4, padding=1)
+        self.conv2 = nn.Conv2d(32, 64, 4, stride=2)
+        self.conv3 = nn.Conv2d(64, 64, 3)
+        self.fc_h_v = NoisyLinear(self.linear_in, args.hidden_size, std_init=args.noisy_std)
```

Comment:
what is fc_h_v, fc_h_a, fc_z_v and fc_z_a? fully connected from h to value/advantage and fully connected from z to value/advantage?
What's Z? 
Can we explain this in one comment?